rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+She generates a non-membership proof that shows her nullifier is not recorded in the tree, and the transaction is allowed to proceed. Once the transaction is completed, 
+the nullifier is added to the tree, ensuring that the UTXO cannot be used again.
+
+**Conclusion**
+The Nullifier Tree is a crucial element of Nescience's security. By recording all consumed UTXOs and ensuring that nullifiers are unique, the tree prevents double-spending 
+and maintains the integrity of the blockchain. The non-membership proof mechanism guarantees that every transaction is validated against the tree. This structure supports 
+the scalability and security of the Nescience architecture, providing a reliable method for verifying the validity of transactions while preventing malicious behavior.
+
+## e) Recursive-Friendly Privacy-Preserving zk-zkVM
+The development of the zk-zkVM in Nescience is a work in progress, as the architecture continues to evolve to support privacy-preserving transactions 
+and efficient zero-knowledge proof generation. The goal of the zk-zkVM is to seamlessly integrate with the Nescience state-separation architecture, 
+ensuring that private transactions remain confidential while allowing the network to verify their validity without compromising privacy.
+
+Currently, we are exploring and testing several existing zkVMs to identify the most suitable platform for our needs. Our focus is on finding a zkVM 
+that not only supports the core features of Nescience, such as state separation and privacy, but also provides the efficiency and scalability required 
+for a decentralized system. Once a suitable zkVM is chosen, we will begin implementing advanced privacy features on top of it, including support for 
+confidential transactions, selective disclosure, and recursive proof aggregation.
+
+The integration of these privacy-preserving features with an existing zkVM will enable Nescience to fully employ its state-separation architecture, 
+ensuring that users can conduct private transactions with robust security and scalability. This approach will allow us to leverage the strengths of 
+proven zkVM technologies while enhancing them with the unique privacy and state-separation capabilities that Nescience requires.
+
+
+- **Privacy-Preserving Features**: At its core, the zk-zkVM is designed with privacy in mind. One of the zk-zkVM’s standout privacy features is **selective disclosure**, 
+which allows users to reveal only specific details of a transaction as needed. For example, a user could disclose the transaction amount while concealing the identities 
+of the participants. The zk-zkVM employs advanced encryption techniques to protect this sensitive data. All transaction data is encrypted before being stored on the blockchain, 
+so even if the data is intercepted, it cannot be deciphered without the appropriate decryption keys. One of the crucial privacy-preserving features is the support 
```

Comment:
Perhaps change "one of the" to "Another of the" as we already said "one of the privacy-preserving features" in this paragraph